src/internal/ppsutil/util.go

```diff
@@ -46,19 +49,85 @@ import (
 	"github.com/pachyderm/pachyderm/v2/src/server/worker/common"
 )
 
+// MustPipelineRcName generates the name of the k8s replication controller that
+// manages a pipeline's workers.  It panics on an error.
+func MustPipelineRcName(pi *pps.PipelineInfo) string {
+	name, err := pipelineRcName(pi.GetDetails().GetK8SNameVersion(), pi.GetPipeline().GetProject().GetName(), pi.GetPipeline().GetName(), pi.GetVersion())
+	if err != nil {
+		panic(err)
+	}
+	return name
+}
+
 // PipelineRcName generates the name of the k8s replication controller that
-// manages a pipeline's workers
-func PipelineRcName(pi *pps.PipelineInfo) string {
-	// k8s won't allow RC names that contain upper-case letters
-	// or underscores
-	//
-	// TODO(CORE-1099): deal with name collision & too-long names
-	pipelineName := strings.ReplaceAll(pi.Pipeline.Name, "_", "-")
-	if projectName := pi.Pipeline.Project.GetName(); projectName != "" {
-		projectName = strings.ReplaceAll(projectName, "_", "-")
-		return fmt.Sprintf("pipeline-%s-%s-v%d", strings.ToLower(projectName), strings.ToLower(pipelineName), pi.Version)
+// manages a pipeline's workers.
+func PipelineRcName(pi *pps.PipelineInfo) (string, error) {
+	return pipelineRcName(pi.GetDetails().GetK8SNameVersion(), pi.GetPipeline().GetProject().GetName(), pi.GetPipeline().GetName(), pi.GetVersion())
+}
+
+func pipelineRcName(nameVersion, projectName, pipelineName string, version uint64) (string, error) {
+	switch nameVersion {
+	case "v2.4":
+		// k8s won't allow RC names that contain upper-case letters
+		// or underscores
+		//
+		// TODO(CORE-1099): deal with name collision & too-long names
+		pipelineName := strings.ReplaceAll(pipelineName, "_", "-")
+		return fmt.Sprintf("pipeline-%s-v%d", strings.ToLower(pipelineName), version), nil
+	case "v2.5":
+		version := strconv.FormatUint(version, 10)
+		charsAvailable := 63 - (len("pipeline---v") + len(version))
+		pipeline := &pps.Pipeline{Project: &pfs.Project{Name: projectName}, Name: pipelineName}
+		h := sha1.New()
```

Comment:
SHA1 will definitely work here, but it doesn't feel great to use a hash function which is no longer secure against collisions.
https://security.googleblog.com/2017/02/announcing-first-sha1-collision.html

 We have the `pachhash` package which is just BLAKE2b, we use that throughout the storage layer.  You can truncate the output.  `pachhash.Sum(input)[:20]`

Reply:
I saw your comment on the design doc.  Given the switch to base32, I think 16 bytes is probably a better compromise.